server/tso/allocator_manager.go

```diff
@@ -288,28 +289,62 @@ func (am *AllocatorManager) campaignAllocatorLeader(parentCtx context.Context, a
 	}
 }
 
-// AllocatorDaemon is used to update every allocator's TSO.
-func (am *AllocatorManager) AllocatorDaemon(serverCtx context.Context) {
+// AllocatorDaemon is used to update every allocator's TSO and check whether we have
+// any new local allocator that needs to be set up.
+func (am *AllocatorManager) AllocatorDaemon(serverCtx context.Context, serverCtxCancel context.CancelFunc) {
 	tsTicker := time.NewTicker(UpdateTimestampStep)
 	defer tsTicker.Stop()
+	checkerTicker := time.NewTicker(checkAllocatorStep)
+	defer checkerTicker.Stop()
 
 	for {
 		select {
 		case <-tsTicker.C:
-			// Filter out allocators without leadership and uninitialized
-			allocatorGroups := am.getAllocatorGroups(FilterUninitialized(), FilterUnavailableLeadership())
-			// Update each allocator concurrently
-			for _, ag := range allocatorGroups {
-				am.wg.Add(1)
-				go am.updateAllocator(ag)
-			}
-			am.wg.Wait()
+			// Update the Local TSO Allocator leaders TSO in memory concurrently.
+			am.allocatorUpdater()
+		case <-checkerTicker.C:
+			// Check if we have any new dc-location configured, if yes,
+			// then set up the corresponding local allocator.
+			am.allocatorChecker(serverCtx, serverCtxCancel)
 		case <-serverCtx.Done():
 			return
 		}
 	}
 }
 
+func (am *AllocatorManager) allocatorChecker(ctx context.Context, ctxCancel context.CancelFunc) {
```

Comment:
What about `patrolAllocatorGroups`. And What if the `dc-location` is unfounded while its allocator is still running. For example, users want to remove a `dc` zone cluster.